src/InsetOrderOptimizer.cpp

```diff
@@ -52,81 +54,54 @@ InsetOrderOptimizer::InsetOrderOptimizer(const FffGcodeWriter& gcode_writer,
 bool InsetOrderOptimizer::addToLayer()
 {
     // Settings & configs:
-    const bool pack_by_inset = ! settings.get<bool>("optimize_wall_printing_order");
-    const InsetDirection inset_direction = settings.get<InsetDirection>("inset_direction");
-    const bool alternate_walls = settings.get<bool>("material_alternate_walls");
+    const auto pack_by_inset = ! settings.get<bool>("optimize_wall_printing_order");
+    const auto inset_direction = settings.get<InsetDirection>("inset_direction");
+    const auto alternate_walls = settings.get<bool>("material_alternate_walls");
 
     const bool outer_to_inner = inset_direction == InsetDirection::OUTSIDE_IN;
-    
-    size_t start_inset;
-    size_t end_inset;
-    int direction;
-    //If the entire wall is printed with the current extruder, print all of it.
-    if (wall_0_extruder_nr == wall_x_extruder_nr && wall_x_extruder_nr == extruder_nr)
-    {
-        //If printing the outer inset first, start with the lowest inset.
-        //Otherwise start with the highest inset and iterate backwards.
-        if(inset_direction == InsetDirection::OUTSIDE_IN)
-        {
-            start_inset = 0;
-            end_inset = paths.size();
-            direction = 1;
-        }
-        else //INSIDE_OUT or CENTER_LAST.
+    const bool use_one_extruder = wall_0_extruder_nr == wall_x_extruder_nr;
+    const bool current_extruder_is_wall_x = wall_x_extruder_nr == extruder_nr;
+
+    auto should_reverse = [&](){
```

Comment:
Functions are variables. Since the function _itself_ doesn't change (even tho' it can change other things) we can make it const :-)

I'll go over the code with a const hammer later.